src/support/json.h

```diff
@@ -403,6 +409,74 @@ struct Value {
     assert(isObject());
     return obj->count(x) > 0;
   }
+
+private:
+  // If the string has no escaped characters, setString() the char* directly. If
+  // it does require escaping, do that and intern a new string with those
+  // contents.
+  void unescapeAndSetString(char* str) {
+    if (!strchr(str, '\\')) {
+      // No escaping slash.
+      setString(str);
+      return;
+    }
+
+    // Otherwise, we may need to escape, so do the work for that.
+    std::vector<char> unescaped;
+    size_t i = 0;
+    while (str[i]) {
+      if (str[i] != '\\') {
+        // Normal character.
+        unescaped.push_back(str[i]);
+        i++;
+        continue;
+      }
+
+      // Escaped character.
+      char c = str[i + 1];
+      if (c != 'u') {
+        switch (c) {
+          case 'b':
+            c = '\b';
+            break;
+          case 'f':
+            c = '\f';
+            break;
+          case 'n':
+            c = '\n';
+            break;
+          case 'r':
+            c = '\r';
+            break;
+          case 't':
+            c = '\t';
+            break;
+        }
+        unescaped.push_back(c);
+        i += 2;
+        continue;
+      }
+
+      // \uXXXX, 4-digit hex number. First, read the hex.
+      unsigned int x;
+      std::stringstream unhex;
+      unhex << std::hex << std::string_view(str + i + 2, 4);
+      unhex >> x;
+
+      // Write out the results.
+      unescaped.push_back(x & 0xff);
+      x >>= 8;
+      if (x) {
+        unescaped.push_back(x);
+      }
+      // TODO UTF stuff
+
+      i += 6;
+    }
+
+    setString(
+      IString(std::string_view(unescaped.data(), unescaped.size()), false));
```

Comment:
We should also update this code to use string_views instead of char* so it can handle `\u0000` in the strings.

Reply:
Sorry, I don't follow you. This should already be handling `\u0000` in the strings, as the test shows - there is a null byte in the string, which gets escaped to that. (Specifically, the input here is escaped JSON, which doesn't have null bytes, only the output can, and we avoid the problem in the output by using a vector of chars and then a string view.)